pinot-core/src/test/java/org/apache/pinot/queries/StatisticalQueriesTest.java

```diff
@@ -0,0 +1,749 @@
+    assertTrue(
+        Precision.equalsWithRelativeTolerance((double) results[3], expectedVariances[3].getResult(), RELATIVE_EPSILON));
+    assertTrue(
+        Precision.equalsWithRelativeTolerance((double) results[4], expectedVariances[4].getResult(), RELATIVE_EPSILON));
+    assertTrue(
+        Precision.equalsWithRelativeTolerance((double) results[5], expectedVariances[5].getResult(), RELATIVE_EPSILON));
+    assertTrue(
+        Precision.equalsWithRelativeTolerance((double) results[6], expectedVariances[6].getResult(), RELATIVE_EPSILON));
+    assertTrue(
+        Precision.equalsWithRelativeTolerance((double) results[7], expectedVariances[7].getResult(), RELATIVE_EPSILON));
+
+    VarianceTuple test = ((VarianceTuple) aggregationResult.get(0));
+    test.apply((new VarianceTuple(0, 0, 0.0d)));
+    System.out.println(test.getM2());
+    // Validate the response for a query with a filter
+    query = "SELECT VAR_POP(intColumnX) from testTable" + getFilter();
+    brokerResponse = getBrokerResponse(query);
+    brokerResponse.getResultTable();
+    results = brokerResponse.getResultTable().getRows().get(0);
+    Variance filterExpectedVariance = new Variance(false);
+    for (int i = 0; i < NUM_RECORDS / 2; i++) {
+      filterExpectedVariance.increment(_intColX[i]);
+    }
+    assertTrue(Precision.equalsWithRelativeTolerance((double) results[0], filterExpectedVariance.getResult(),
+        RELATIVE_EPSILON));
+  }
+
+  @Test
+  public void testVarianceAggregationGroupBy() {
+    // Compute expected group results
+    Variance[] expectedGroupByResult = new Variance[NUM_GROUPS];
+    double[] expectedSum = new double[NUM_GROUPS];
+
+    for (int i = 0; i < NUM_GROUPS; i++) {
+      expectedGroupByResult[i] = new Variance(false);
+    }
+    for (int j = 0; j < NUM_RECORDS; j++) {
+      int pos = j / (NUM_RECORDS / NUM_GROUPS);
+      expectedGroupByResult[pos].increment(_intColX[j]);
+      expectedSum[pos] += _intColX[j];
+    }
+
+    String query = "SELECT VAR_POP(intColumnX) FROM testTable GROUP BY groupByColumn ORDER BY groupByColumn";
+    GroupByOperator groupByOperator = getOperator(query);
+    GroupByResultsBlock resultsBlock = groupByOperator.nextBlock();
+    QueriesTestUtils.testInnerSegmentExecutionStatistics(groupByOperator.getExecutionStatistics(), NUM_RECORDS, 0,
+        NUM_RECORDS * 2, NUM_RECORDS);
+    AggregationGroupByResult aggregationGroupByResult = resultsBlock.getAggregationGroupByResult();
+    assertNotNull(aggregationGroupByResult);
+    for (int i = 0; i < NUM_GROUPS; i++) {
+
+      VarianceTuple actualVarianceTuple = (VarianceTuple) aggregationGroupByResult.getResultForGroupId(0, i);
+      checkWithPrecisionForVariance(actualVarianceTuple, NUM_RECORDS / NUM_GROUPS, expectedSum[i],
+          expectedGroupByResult[i].getResult(), false);
+    }
+  }
+
+  @Test
+  public void testStandardDeviationAggregationOnly() {
```

Comment:
saw some instability on https://github.com/apache/pinot/actions/runs/3652424011/jobs/6170799903.

Reply:
Should be fixed with #9948